src/SignhostAPIClient/Rest/SignHostApiClient.cs

```diff
@@ -21,6 +21,32 @@ public class SignHostApiClient
 		private readonly ISignHostApiClientSettings settings;
 		private readonly HttpClient client;
 
+        /// <summary>
+        /// Calculates and validates checksum.
+        /// </summary>
+        /// <param name="transactionId">Transaction identifier.</param>
+        /// <param name="status">Status.</param>
+        /// <param name="sharedSecret">Shared secret.</param>
+        public void CalculateAndValidateChecksum(                               //has to become private
+                                        string transactionId,
+                                        string transactionStatus,
+                                        string sharedSecret,
+                                        string validChecksum) {
+            string calculatedChecksum;
+            using (var sha1 = System.Security.Cryptography.SHA1.Create()) {
+                var x = $"{transactionId}||{transactionStatus}|{sharedSecret}";
+                var byteArray = System.Text.Encoding.UTF8.GetBytes(x);
+                byte[] hash = sha1.ComputeHash(byteArray);
+                calculatedChecksum = System.BitConverter
+                                           .ToString(hash)
+                                           .Replace("-", string.Empty)
+                                           .ToLower();
+            };
```

Comment:
A closing `}` should not have a ;,
After a closing `}` there should be a empty newline (unless followed by another `}`'